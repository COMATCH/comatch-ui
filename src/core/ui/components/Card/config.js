const BACKGROUND_COLOR = {
    BLACK: 'black',
    WHITE: 'white',
    LIGHT_BLUE: 'lightBlue',
    PRIMARY: 'primary',
    LIGHT_PRIMARY: 'lightPrimary',
    SECONDARY: 'secondary',
    LIGHT_GRAY: 'lightGray',
    MID_GRAY: 'midGray',
    GRAY: 'gray',
    DARK_GRAY: 'darkGray',
    GREEN: 'green',
    LIGHT_GREEN: 'lightGreen',
    DARK_GREEN: 'darkGreen',
    RED: 'red',
    LIGHT_RED: 'lightRed',
    DARK_RED: 'darkRed',
    LIGHT_ORANGE: 'lightOrange',
    ORANGE: 'orange',
    DARK_ORANGE: 'darkOrange',
    YELLOW: 'yellow',
    TRANSPARENT: 'transparent',
};

const FONT_COLOR = {
    BLACK: 'white',
    WHITE: 'black',
    LIGHT_BLUE: 'black',
    PRIMARY: 'white',
    LIGHT_PRIMARY: 'black',
    SECONDARY: 'white',
    LIGHT_GRAY: 'black',
    MID_GRAY: 'black',
    GRAY: 'white',
    DARK_GRAY: 'white',
    GREEN: 'white',
    LIGHT_GREEN: 'black',
    DARK_GREEN: 'white',
    RED: 'white',
    LIGHT_RED: 'black',
    DARK_RED: 'white',
    LIGHT_ORANGE: 'black',
    ORANGE: 'black',
    DARK_ORANGE: 'white',
    YELLOW: 'black',
    TRANSPARENT: 'black',
};

const ORIENTATION = {
    HORIZONTAL: 'horizontal',
    VERTICAL: 'vertical',
};

export { BACKGROUND_COLOR, FONT_COLOR, ORIENTATION };
