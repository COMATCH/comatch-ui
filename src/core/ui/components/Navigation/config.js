export const ORIENTATION = {
    HORIZONTAL: 'horizontal',
    VERTICAL: 'vertical',
    RESPONSIVE: 'responsive',
};
