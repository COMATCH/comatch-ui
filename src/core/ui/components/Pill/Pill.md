## Pill

prop | type | default | required | description
---- | :----: | :-------: | :--------: | -----------
**children** | `ReactNode` | `null` | :x: | 
**className** | `String` | `null` | :x: | 
**id** | `String` | `null` | :x: | 
**size** | `Enum(Object.values(SIZES))` | `null` | :x: | 
**type** | `Enum(Object.values(TYPES))` | `null` | :x: | 

