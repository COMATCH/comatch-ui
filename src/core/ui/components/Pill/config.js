import { palette, typography } from '../../../styles/variables';

const SIZES = {
    XS: 'XS',
    SM: 'SM',
    MD: 'MD',
    LG: 'LG',
    XL: 'XL',
    XXL: 'XXL',
};

const SIZES_IN_PIXELS = {
    XS: {
        fontSize: typography.fontXs,
        lineHeight: typography.lineHeightXs,
    },
    SM: {
        fontSize: typography.fontSm,
        lineHeight: typography.lineHeightSm,
    },
    MD: {
        fontSize: typography.fontMd,
        lineHeight: typography.lineHeightMd,
    },
    LG: {
        fontSize: typography.fontLg,
        lineHeight: typography.lineHeightLg,
    },
    XL: {
        fontSize: typography.fontXl,
        lineHeight: typography.lineHeightXl,
    },
    XXL: {
        fontSize: typography.fontXxl,
        lineHeight: typography.lineHeightXxl,
    },
};

const TYPES = {
    DISABLED: 'DISABLED',
    ERROR: 'ERROR',
    INFO: 'INFO',
    PRIMARY: 'PRIMARY',
    SECONDARY: 'SECONDARY',
    SUCCESS: 'SUCCESS',
    WARN: 'WARN',
};

const TYPES_IN_COLOR = {
    DISABLED: {
        background: palette.lightGray,
        borderColor: palette.lightGray,
        color: palette.black,
    },
    ERROR: {
        background: palette.red,
        borderColor: palette.red,
        color: palette.white,
    },
    INFO: {
        background: palette.lightBlue,
        borderColor: palette.lightBlue,
        color: palette.black,
    },
    PRIMARY: {
        background: palette.primary,
        borderColor: palette.primary,
        color: palette.white,
    },
    SECONDARY: {
        background: palette.secondary,
        borderColor: palette.secondary,
        color: palette.white,
    },
    SUCCESS: {
        background: palette.green,
        borderColor: palette.green,
        color: palette.white,
    },
    WARN: {
        background: palette.orange,
        borderColor: palette.orange,
        color: palette.white,
    },
};

export { SIZES, SIZES_IN_PIXELS, TYPES, TYPES_IN_COLOR };
