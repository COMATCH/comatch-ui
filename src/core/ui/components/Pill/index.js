export { SIZES as PILL_SIZES, TYPES as PILL_TYPES } from './config';
export * from './Pill';
