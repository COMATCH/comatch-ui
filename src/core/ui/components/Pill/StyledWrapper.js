import React, { forwardRef } from 'react';
import styled from 'styled-components';
import { SIZES_IN_PIXELS, TYPES_IN_COLOR } from './config';
import { palette } from '../../../styles/variables';

function generateStyling({ size, type }) {
  const colorSpecs = TYPES_IN_COLOR[type] || {
    background: 'transparent',
    borderColor: palette.black,
    color: palette.black,
  };
  const fontSpecs = SIZES_IN_PIXELS[size] || SIZES_IN_PIXELS.XS;

  return `
    background-color: ${colorSpecs.background};
    border: 1px solid ${colorSpecs.borderColor};
    color: ${colorSpecs.color};
    font-size: ${fontSpecs.fontSize};
    line-height: ${fontSpecs.lineHeight};
  `;
}

const Node = forwardRef(({ size, type, ...rest }, ref) => <div {...rest} ref={ref} />);
const StyledWrapper = styled(Node)`
  align-content: center;
  border-radius: 15px;
  display: inline-flex;
  font-weight: bold;
  justify-content: center;
  overflow: hidden;
  padding: 5px 10px;

  ${generateStyling}
`;

export { StyledWrapper };
