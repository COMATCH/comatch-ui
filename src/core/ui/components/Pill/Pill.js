import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { SIZES, TYPES } from './config';
import { StyledWrapper } from './StyledWrapper';

const propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    id: PropTypes.string,
    size: PropTypes.oneOf(Object.values(SIZES)),
    type: PropTypes.oneOf(Object.values(TYPES)),
};

const defaultProps = {
    children: null,
    className: null,
    id: null,
    size: null,
    type: null,
};

const Pill = ({ children, className, id, size, type }) => !!children && (
    <StyledWrapper
        className={classnames('Pill', className)}
        {...(id && { id })}
        size={size}
        type={type}
    >
        {children}
    </StyledWrapper>
);

Pill.propTypes = propTypes;
Pill.defaultProps = defaultProps;

export { Pill };
