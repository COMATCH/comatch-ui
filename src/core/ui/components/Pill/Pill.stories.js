import React from 'react';
import styled from 'styled-components';
import Highlight from 'react-highlight';
import { storiesOf } from '@storybook/react';
import { Pill } from './Pill';
import { SIZES, TYPES } from './config';

const PillCollection = styled.div`
    .Pill ~ .Pill {
        margin: 10px;
    }
`;

storiesOf('Pill', module)
    .add('empty', () => (
        <>
            <Pill />
            <Highlight className="html">{'<Pill />'}</Highlight>
        </>
    ))
    .add('sizes', () => (
        <>
            <PillCollection>
                <Pill>Hello world</Pill>
                {Object.values(SIZES).map((size) => (
                    <Pill key={size} size={size}>
                        Hello world
                    </Pill>
                ))}
            </PillCollection>
            <Highlight className="html">
                {Object.values(SIZES).reduce(
                    (acc, size) => `${acc}\n<Pill size={PILL_SIZES.${size}}>Hello world</Pill>`,
                    '<Pill>Hello world</Pill>',
                )}
            </Highlight>
        </>
    ))
    .add('types', () => (
        <>
            <PillCollection>
                <Pill>Hello world</Pill>
                {Object.values(TYPES).map((type) => (
                    <Pill key={type} type={type}>
                        Hello world
                    </Pill>
                ))}
            </PillCollection>
            <Highlight className="html">
                {Object.values(TYPES).reduce(
                    (acc, type) => `${acc}\n<Pill type={PILL_TYPES.${type}}>Hello world</Pill>`,
                    '<Pill>Hello world</Pill>',
                )}
            </Highlight>
        </>
    ))
    .add('sizes & types', () => (
        <>
            <PillCollection>
                {[null, ...Object.values(TYPES)].map((type) => (
                    <div key={type}>
                        {[null, ...Object.values(SIZES)].map((size) => (
                            <Pill key={`${type}--${size}`} size={size} type={type}>
                                Hello world
                            </Pill>
                        ))}
                    </div>
                ))}
            </PillCollection>
            <Highlight className="html">
                {[null, ...Object.values(TYPES)].reduce(
                    (acc, type, index) =>
                        `${acc}${index ? '\n' : ''}<div>\n${[null, ...Object.values(SIZES)].reduce(
                            (subAcc, size, subIndex) =>
                                `${subAcc}${subIndex ? '\n' : ''}\t<Pill${size ? ` size={PILL_SIZES.${size}}` : ''}${
                                    type ? ` type={PILL_TYPES.${type}}` : ''
                                }>Hello world</Pill>`,
                            '',
                        )}\n</div>`,
                    '',
                )}
            </Highlight>
        </>
    ));
