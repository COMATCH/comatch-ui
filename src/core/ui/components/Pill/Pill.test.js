import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { SIZES, TYPES } from './config';
import { Pill } from './Pill';
import '../../../config/tests/setup';

describe('Pill', () => {
    it('should render the default Pill correctly', () => {
        const pill = renderer.create(<Pill>Hello World</Pill>).toJSON();
        expect(pill).toMatchSnapshot();
    });

    it('should have the correct default props', () => {
        const pill = mount(<Pill />);
        expect(pill.prop('className')).toEqual(null);
        expect(pill.prop('id')).toEqual(null);
        expect(pill.prop('size')).toEqual(null);
        expect(pill.prop('type')).toEqual(null);
    });

    describe('PILL_SIZES & PILL_TYPES mapping', () => {
        [null, ...Object.values(TYPES)].forEach((type) => {
            [null, ...Object.values(SIZES)].forEach((size) => {
                const pill = mount((
                    <Pill className="Test" id="test" size={size} type={type}>
                        Hello world
                    </Pill>
                ));

                expect(pill.prop('className')).toEqual('Test');
                expect(pill.prop('id')).toEqual('test');
                expect(pill.prop('size')).toEqual(size);
                expect(pill.prop('type')).toEqual(type);
            });
        });        
    });
});
