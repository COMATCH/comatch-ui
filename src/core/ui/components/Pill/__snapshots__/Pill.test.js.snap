// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill should render the default Pill correctly 1`] = `
<div
  className="Pill sc-bdVaJa jySjDV"
>
  Hello World
</div>
`;
